### DISCUSSION TITLE
Presentations/Notebooks etc in a book project

### DISCUSSION THREAD

**Discussion by kdschneider** [quarto-dev/quarto-cli#3194] [Q&A]
Hello,

I am trying to write my masters thesis using quarto.

Part of my thesis are presentations and notebooks that are not part of the books chapters, but still those have to be rendered when rendering the book.

Putting them in their own projects is not possible, as book, presentations and notebooks share data, plots and citations.

Ideally quarto should first render the notebooks, then the book and presentations. Is this somehow possible?

**Comment by mcanouil** [comment]
I don't understand what's the structure of your "project", I would prefixed the files not to be rendered as part of the book with "_" which will avoid Quarto to render them within the book.
Then I would write a pre-render script which basically call `quarto render _some-presentation.qmd`.
See <https://quarto.org/docs/projects/scripts.html#pre-and-post-render> for details.

```yaml
project:
  pre-render:
    - render-some-quarto-docs-first-before-book.sh
```

You could probably also use "project profiles" to handle the different type of documents in your thesis, see <https://quarto.org/docs/projects/profiles.html>.

**Reply by kdschneider** [reply]
Thanks for your tips. I currently use the pre-render strategy.

My project structure is:

- masters thesis
  - thesis (chapters + tex stuff)
  - slides
  - notebooks
  - data
  - figures

Project profiles sound like a nice solution. Unfortunately the current homebrew version is 1.1.251 and profiles are not yet available...

**Reply by mcanouil** [reply]
You can install directly Quarto without homebrew or create your own cask, see <https://github.com/quarto-dev/quarto-cli/discussions/2955#discussioncomment-3928804>.

**Reply by kdschneider** [reply]
Ah nice. I will try that tomorrow. Thanks for your help.

**Reply by tomkom** [reply]
IS there a chance you could share your setup (some github or so) I would like to replicate exactly this. Much appreciated! Currently I have two separate _quarto.yml files, that I have to keep synced, and I have to rename one, render slides, then another. LInking from the book/website to the slides is rather hard.

**Reply by mcanouil** [reply]
@tomkom Project profile is what you need to use, see <https://quarto.org/docs/projects/profiles.html>.
